KSF-20_toDoList

Madridka opened the pull request.

Madridka committed:
edit 1 error

Madridka committed:
-added taskinput.vue
-added tasklist.vue

Madridka committed:
delete file completedlist.vue

Madridka committed:
add taskitem.vue

Madridka committed:
trouble:
how to link TaskItem.vue and TaskList.vue

Madridka committed:
fix: link TaskList and TaskItem, by props drilling, but now i have error: "Unexpected mutation of "task" prop  vue/no-mutating-props"

refactor: move methods with filter of tasks (comp, incomp) to App.vue

Madridka committed:
fix: change v-model in TaskItem (no-mutanting-vue) to :checked, and @change
refactor: rename aaa in styles.css, and other similar class
fix: add class "completed-tasks" to TaskList

Madridka committed:
feat: add edit button with functions

Madridka committed:
feat: all functions is ready (ready?)

next station is trying to refactor all styles

Madridka committed:
bug: text-decoration: line-through; isnt working?
but text-transform: uppercase; is working at the same li

Maiev93 commented on src/App.vue:
```diff
@@ -2,30 +2,35 @@
+        @update-task="updateTask"
+      />
+      <TaskList
+        class="completed-tasks"
```
Лучше класс писать в компоненте, а не в родителе, в родителе можно писать его, если нужна стилизация компонента внутри родителя. возможно, из-за этого отваливается стиль text-decor: line-through. также не забывай про scoped компонентов.

Madridka replied:
здесь я оставлю так, потому что при отрисовке компонента еще передаются пропсы :tasks="tasksComplete", для отрисовки выполненных задач, и отделения их логически и тайтлом от невыполненных
text-decor: l-t починил в следующих коммитах

Maiev93 commented on src/assets/style.css:
```diff
@@ -5,23 +5,23 @@ body {
 }
 
-.aaa {
+.container {
```
Здесь я вообще имела ввиду, что стили изначально хоть и будут использоваться те же, но разнести их по компонентам все равно нужно. в отдельном файле .css пишутся обычно стили reset/normalise, переменные, если необходимы, глобальные стили.

Madridka replied:
понял, учел
раскидал стили на base.scss, normalize.scss 
а так же поубирал стили в компоненты в scoped

Maiev93 commented on src/assets/style.css:
```diff
@@ -151,14 +151,15 @@ button.delete img:hover {
 /* Edit Task */
 ul li input[type=text] {
-    display: none
+    /* display:none */
```
Лишние комментарии необходимо убирать, только если не требуется доработка в будущем. если нужна доработка, сверху пишется комментарий // TODO: need some work...

Madridka replied:
убрал
эти комментарии были все в ходе тестирования, сейчас поубирал лишнее

Maiev93 commented on src/components/TaskItem.vue:
```diff
@@ -0,0 +1,89 @@
+      
+    },
+    saveEdit() {
+      console.log("TaskItem saveEdit:", this.localEditTask);
```
Ненужные логи, которые пригождаются во время разработки, необходимо убирать

Madridka replied:
убрано

Maiev93 commented on src/components/TaskItem.vue:
```diff
@@ -0,0 +1,89 @@
+    },
+  },
+
+  watch: {
```
Некорретный порядок опций, в документации Vue есть правила описания опций, там есть другие правила для компонентов, они носят рекомендательный характер, но лучше их придерживаться. должен быть порядок
  - data
  - computed
  - watch
  - lifecycle hooks
  - methods

Madridka replied:
нашел почитал на русском про "Порядок опций в компоненте/экземпляре"
на vue 3 правда, но тоже сойдет
переделал все компоненты

Maiev93 commented on src/App.vue:
```diff
@@ -37,19 +42,73 @@ import TaskList from "./components/TaskList.vue";
+      this.tasks = this.tasks.filter((task) => task.id !== id);
+    },
+  },
+  computed: {
```
Неверный порядок опций, см. watch

Madridka replied:
исправил

Madridka committed:
fix: add alt for image
fix: кнопки сохранить и отменить наезжают друг на друга
style: completed tasks

Maiev93 commented on src/App.vue:
```diff
@@ -2,30 +2,35 @@
+      />
+      <TaskList
+        :tasks="tasksComplete"
+        class="completed-tasks"
```
Лучше класс писать в компоненте, а не в родителе, в родителе можно писать его, если нужна стилизация компонента внутри родителя. возможно, из-за этого отваливается стиль text-decor: line-through. также не забывай про scoped компонентов.

Madridka committed:
fix: order of options in the component

Madridka committed:
feat: install sass loader
style: create base.scss, normalize.scss
style: all specific styles move to component in style scoped
